model/project.go

```diff
@@ -136,7 +136,8 @@ type BuildVariantTaskUnit struct {
 	// with BatchTime and CronBatchTime being mutually exclusive.
 	CronBatchTime string `yaml:"cron,omitempty" bson:"cron,omitempty"`
 	// If Activate is set to false, then we don't initially activate the task.
-	Activate *bool `yaml:"activate,omitempty" bson:"activate,omitempty"`
+	Activate       *bool `yaml:"activate,omitempty" bson:"activate,omitempty"`
+	RunOnContainer *bool `yaml:"run_on_container,omitempty" bson:"run_on_container"`
```

Comment:
Rather than having a user manually set this field, would be it better to have it automatically detect the container vs. distro configuration based on what they've passed to `RunOn`? Otherwise, they'll have to mnnually set/unset this if they change `RunOn` between distros and containers.

For example, if they pass one name to `RunOn`, it checks if there's a matching container name and uses that if possible, but falls back to using a distro otherwise. If there are multiple names, they're assumed to refer to distros.

Reply:
This wasnt intended to be a manually set field, but something that gets derived and set during the parsing process. However I have changed it to only setting ExecutionPlatform based off inspection of RunOn now, so this property has been removed.